tabs-tabs-tabs/tabs.js

```diff
@@ -0,0 +1,42 @@
+document.addEventListener("click", function(e) {
+  function callOnActiveTab(callback) {
+    chrome.tabs.query({}, function(tabs) {
+      for (let tab of tabs) {
+        if (tab.active) {
+          callback(tab);
+        }
+      }
+    });
+  }
+
+  if (e.target.id === "tabs-move-beginning") {
+    callOnActiveTab((tab) => {
+      chrome.tabs.move([tab.id], {index: 0});
+    });
+  }
+
+  if (e.target.id === "tabs-move-end") {
+    callOnActiveTab((tab) => {
+      chrome.tabs.move([tab.id], {index: -1});
+    });
+  }
+
+  else if (e.target.id === "tabs-duplicate") {
+    callOnActiveTab((tab) => {
+      chrome.tabs.duplicate(tab.id);
```

Comment:
Well its just about to https://bugzilla.mozilla.org/show_bug.cgi?id=1238312, I wrote this to help test the API.